### ISSUE TITLE
No database waiting time.

### ISSUE THREAD

**Issue by mariushosting** [linuxserver/docker-bookstack#159]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The current bookstack image have no database waiting time. So you need to restart the container to be able to connect to the database.

### Steps To Reproduce

The current bookstack image have no database waiting time. So you need to restart the container to be able to connect to the database.

### Environment

```markdown
- OS:
- How docker service was installed:
```

### CPU architecture

x86-64

### Docker creation

```bash
version: "3.9"
services:
  bookstack:
    image: ghcr.io/linuxserver/bookstack
    container_name: Bookstack
    healthcheck:
      test: curl -f http://localhost:80/ || exit 1
    environment:
      - TZ=Europe/Bucharest
      - PUID=1026
      - PGID=100 
      - APP_URL=http://YourNASIP:6875
      - DB_HOST=bookstack_db
      - DB_USER=bookstack
      - DB_PASS=yourdbpass
      - DB_DATABASE=bookstackapp
    volumes:
      - /volume1/docker/bookstack:/config
    ports:
      - 6875:80
    restart: on-failure:5
    depends_on:
      - bookstack_db
  bookstack_db:
    image: mysql
    container_name: Bookstack-DB
    healthcheck:
      test: out=$$(mysqladmin ping -h localhost -P 3306 -u root --password=yourdbpass 2>&1); echo $$out | grep 'mysqld is alive' || { echo $$out; exit 1; }
    environment:
      - TZ=Europe/Bucharest
      - MYSQL_ROOT_PASSWORD=yourdbpass
      - MYSQL_DATABASE=bookstackapp
      - MYSQL_USER=bookstack
      - MYSQL_PASSWORD=yourdbpass
    volumes:
      - /volume1/docker/bookstackdb:/var/lib/mysql:rw
    restart: always
```

### Container logs

```bash
Waiting for DB to be available
   Illuminate\Database\QueryException 
  SQLSTATE[HY000] [2002] Connection refused (SQL: select * from information_schema.tables where table_schema = bookstackapp and table_name = migrations and table_type = 'BASE TABLE')
  at /app/www/vendor/laravel/framework/src/Illuminate/Database/Connection.php:712
    708▕         // If an exception occurs when attempting to run a query, we'll format the error
    709▕         // message to include the bindings with SQL, which will make this exception a
    710▕         // lot more helpful to the developer instead of just the database's errors.
    711▕         catch (Exception $e) {
  ➜ 712▕             throw new QueryException(
    713▕                 $query, $this->prepareBindings($bindings), $e
    714▕             );
    715▕         }
    716▕     }
      +36 vendor frames 
  37  /app/www/artisan:37
      Illuminate\Foundation\Console\Kernel::handle()
[custom-init] No custom files found, skipping...
[ls.io-init] done.
```

**Comment by ssddanbrown** [comment]
I have not used a super-recent version of the image, but from memory it would wait, and I could bring everything up at once.
Looks like [this would be](https://github.com/linuxserver/docker-bookstack/blob/6aaf90dff80b797bd094d53b83a1233617cd0a32/root/etc/s6-overlay/s6-rc.d/init-bookstack-config/run#L119-L135) the relevant part of the script.

Do you have positive confirmation that this is timing based? 
Just asking since the log error provided is not clear in that.
Does everyone come up fine when you specifically start the database first?

**Comment by mariushosting** [comment]
> I have not used a super-recent version of the image, but from memory it would wait, and I could bring everything up at once. Looks like [this would be](https://github.com/linuxserver/docker-bookstack/blob/6aaf90dff80b797bd094d53b83a1233617cd0a32/root/etc/s6-overlay/s6-rc.d/init-bookstack-config/run#L119-L135) the relevant part of the script.
> 
> Do you have positive confirmation that this is timing based? Just asking since the log error provided is not clear in that. Does everyone come up fine when you specifically start the database first?

The linuxserver bookstack image immediately tries to connect to the database before the database starts and this cause the issue. I just switch to the **solidnerd** bookstack image to solve the issue. Solidnerd have a db waiting time set before connecting to the database.

**Comment by aptalca** [comment]
You responded to and quoted a post that links to the piece of code that waits for the database and has an echo line `Waiting for DB to be available`. Really?

Anyway. . .

It would be nice to see a full timestamped log to see if it waited and how long, as it printed that it was waiting.

**Comment by ssddanbrown** [comment]
Okay. For what it's worth, I just tested a fresh compose setup, using the latest images and using the compose setup provided in the readme here, and it all comes up fine. 

I then did a further test by only raising the bookstack container via `docker-compose up --no-deps bookstack`. 
The startup script would wait here:

```
bookstack       | Waiting for DB to be available
bookstack       | nc: bad address 'bookstack_db'
bookstack       | nc: bad address 'bookstack_db'
```

Then I'd start the DB container after a few seconds and it would then continue once the DB was up.
So all looks functional on my side. (Ubuntu 22.10 / Ubuntu 22.10 / docker-compose version 1.29.2).
